neurom/core/_neuron.py

```diff
@@ -131,7 +206,7 @@ def transform(self, trans):
 
     def iter_sections(self, order=Tree.ipreorder):
         '''iteration over section nodes'''
-        return order(self.root_node)
+        return iter_sections(self, iterator_type=order)
```

Comment:
Ok, that is a big change in how the API works: moving away from bare functions for iteration to object style iteration.  I'm not married to either concept, and we can concurrently support them, but there are benefits to bare functions, and hinting towards using them (by not having object accessible iterators) has benefits.

Are we comfortable making that switch?

Also, that is independent of this PR, so if you're happy with it, let me know.
Finally, #558 iter_segments now takes a Section(), so that should be closed when this is closed.